crates/driver/src/domain/mempools.rs

```diff
@@ -102,13 +103,16 @@ impl Mempools {
 
         // Wait for the transaction to be mined, expired or failing.
         let result = async {
-            loop {
+            while let Some(block) = block_stream.next().await {
                 // Wait for the next block to be mined or we time out.
-                if tokio::time::timeout_at(deadline, block_stream.next())
-                    .await
-                    .is_err()
-                {
-                    tracing::info!(?hash, "tx not confirmed in time, cancelling");
+                if block.number > submission_deadline {
```

Comment:
I think at the moment this cancels the tx one block later than it could.
Because the deadline gets checked before checking the blockchain if our tx was mined this check currently has to happen 1 block after the deadline. If this check were to move into the `TxStatus::Pending` case of the `match` below we could already abort if `block.number == submission_deadline`.